### ISSUE TITLE
sam deploy fails with PermissionError when trying to write to  `/.aws-sam`

### ISSUE THREAD

**Issue by LaffeyChris** [aws/aws-sam-cli#3855]
### Description:
When running `sam deploy` (v1.47.0), I encounter a `PermissionError`, as sam is trying to write to `/.aws-sam`. 

As I am running `sam deploy` on a shared CI/CD instance, I do not have access to write within the root directory. 
Looks like the issue is happening when setting up config/telemetry data. 

I traced this back to the last working version (v1.33.0), and it seems that the OS level errors are no longer caught, as is done here: 
https://github.com/aws/aws-sam-cli/blob/v1.33.0/samcli/cli/global_config.py#L80

### Steps to reproduce:
Run `sam deploy` on an instance that does not have permissions to write to `/.aws-sam` dir.

### Observed result:

```
Traceback (most recent call last):
  File "/usr/local/bin/sam", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 73, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 151, in wrapped
    metric = Metric(metric_name)
  File "/usr/local/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 318, in __init__
    self._add_common_metric_attributes()
  File "/usr/local/lib/python3.9/site-packages/samcli/lib/telemetry/metric.py", line 341, in _add_common_metric_attributes
    self._data["installationId"] = self._gc.installation_id
  File "/usr/local/lib/python3.9/site-packages/samcli/cli/global_config.py", line 351, in installation_id
    self.set_value(DefaultEntry.INSTALLATION_ID, value)
  File "/usr/local/lib/python3.9/site-packages/samcli/cli/global_config.py", line 273, in set_value
    self._set_value(config_entry, value, is_flag, flush)
  File "/usr/local/lib/python3.9/site-packages/samcli/cli/global_config.py", line 294, in _set_value
    self._write_config()
  File "/usr/local/lib/python3.9/site-packages/samcli/cli/global_config.py", line 324, in _write_config
    self.config_dir.mkdir(mode=0o700, parents=True, exist_ok=True)
  File "/usr/local/lib/python3.9/pathlib.py", line 1323, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/.aws-sam'
```

### Expected result:

One of the following should happen in place of throwing a `PermissionsError`: 
1. Catch the exception & pass (as previously done)
2. Write to user specified dir (that they have permissions to write to, i.e. dest dir specified in --template-file)

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Inside container - python:3.9.9-slim
3. `sam --version`: v1.47.0
4. AWS region: N/A

**Comment by jfuss** [comment]
@LaffeyChris Thanks for the report. Looks like this was removed with Accelerate's beta launch: https://github.com/aws/aws-sam-cli/pull/3386. It's not clear why, so touching base with the team on this but I think the exception handling needs to be put back (likely was there to cover this exact thing).

**Comment by hawflau** [comment]
Released in SAM CLI v1.49.0. Closing
